orth

```diff
@@ -167,13 +167,16 @@ analyze_in_docker() {
   analyzer_input_dir=$(realpath "$1")
   analyzer_output_dir=$(realpath .)
 
-  docker_image="registry.gitlab.com/oss-review-toolkit/ort-gitlab-ci/ort:latest"
-
-  docker login registry.gitlab.com \
-    --username gitlab-ci-token \
-    --pasword $gitlab_token
+  if [ -n "$ort_docker_registry_server" ] && \
+     [ -n "$ort_docker_registry_username" ] && \
+     [ -n "$ort_docker_registry_password" ]
+  then
+    docker login $ort_docker_registry_server \
+      --username $ort_docker_registry_username \
+      --password $ort_docker_registry_password
+  fi
```

Comment:
For convenience maybe add an `else` block which echoes a message like "Not logging in to Docker registry as credentials are incomplete.".

Reply:
I'd prefer to avoid debug logging for now, to keep things a bit minimal. In the unhappy path an error is output anyway by the `docker` pull command.